### ISSUE TITLE
Mackchainladder's standard error has different estimates compared to R's ChainLadder library

### ISSUE THREAD

**Issue by kmonts** [casact/chainladder-python#234]
I tried the Mackchainladder method with chainladder-python, R-ChainLadder, and Excel using RAA data, a Paid Claims triangle, and an Incurred Claims Triangle (with subrogation that is based on the Paid Claims Triangle). 
The ultimate values are the same in all cases but the Mack Standard Error values are different in Python, while R and excel gives me the same values. What could be the source of the difference?
The triangles I used passes the calendar year test but fails the development factor correlation test. I'm using Chainladder 0.8.9. 
I've uploaded the code, data, and the results from R and Excel [here](https://drive.google.com/drive/folders/1gtfYu1P7CjbsNGjoydrCJZ0CpidISTuy?usp=sharing).

The total Mack Standard Errors of the data are as follows:
Paid Claims
Python: 101,179,700.00 
R: 31,705,239
Excel: 31,705,240 

Incurred Claims
Python: 16,813,400.00 
R: 16,760,616.96
Excel: 16,760,617

RAA
Python:26,881.09
R: 26,909.01

Hoping to know the source of the difference (or maybe this is the expected output) and if there are any adjustments I could make. I've had a lot of fun learning about reserving with this package. Thank you guys so much.

**Comment by jbogaardt** [comment]
Welcome @kmonts,

First, the easy one. RAA between python and R should produce identical results. The difference you're seeing is a consequence of how "sigma" is being estimated. For the R package you're explicitely declaring 'Mack' method.  In python, absent any intervention, it will use 'log-linear' interpolation. You can use mack estimation in python as follows:
```python
import chainladder as cl
dev = cl.Development(sigma_interpolation='mack').fit_transform(cl.load_sample('raa'))
cl.MackChainladder().fit(dev).total_mack_std_err_
```
The Mack method in python is broken up between the `Development`, `TailCurve`, and `MackChainladder` esimtators.  The `Development` estimator is where the development factors are estimated along with corresponding regression statistics of those patterns. Consequently, if we want to atler how the `ldf_` regressions are fit, we need to do it there.


The harder question - your Paid data.  You have no variability in your data from age 84 and later. I think this is why you're seeing `CV(IBNR)` as `NaN` and `Inf` in R.  Python is struggling with this too. I think I need to review how the variability estimates, `sigma_` and `std_err_` are calculated when there is no variability.
```python
tri = cl.Triangle(df, origin='origin', development='development', columns='Paid')
dev = cl.Development(sigma_interpolation='mack').fit(tri)
print(dev.sigma_)
print(dev.std_err_)
```
These should be directly comparable to the R estimates of the same, but are in fact different in the area where variability is missing:
```R
mack1 <- MackChainLadder(tri, est.sigma="Mack")
print(mack1$sigma)
print(mack1$f.se)
```
These differences are propagating through to the final differences you're seeing.  I'll have to think about this more.

**Comment by kmonts** [comment]
Hi @jbogaardt. Thanks and sorry for the late reply. I tried your suggestion in Python and it worked for the RAA dataset. I've verified that for my data- R and Python do have a hard time due to the non-variation in the 84th month, and I've also compared the sigmas computed by both. Good luck and looking forward to your thoughts. Thanks again!

**Comment by jbogaardt** [comment]
I've finally had a chance to review this @kmonts.  The fundamental difference between `chainladder-python` and R's `ChainLadder` is how `sigma_` and `std_err_` (variability statistics) are estimated, as mentioned above.  Because there is no variability in your data for ages 84 and later, the R library assumes zero variability.  The `chainladder-python` package assumes the variability statistics cannot be calculated and uses either 'mack' or 'log-linear' interpolation depending on whichever you've selected to come up with some non-zero variability statistics.

In these cases, one could assume (with an IBNR of 0), that all claims are settled and there is no variability. This does produce considerably lower `total_mack_std_err_` than is currently being produced.  I think it makes sense.  I've added this to the `chainladder-python` package to make it consistent with R's `ChainLadder`  and it will be available in the `0.8.10` release.  This release should be out within a week or so, but if you want to play with it ahead of that, you can install from git:

`pip install git+https://github.com/casact/chainladder-python/`

For what its worth, this paid triangle doesn't play well with the Mack method. Even in R, the moment you add on a tail, the whole setup fails:
```R
mack1 <- MackChainLadder(tri, tail=TRUE)
```

**Comment by kmonts** [comment]
Thank you very much for all the work and for the clarifications. Looking forward to the release!
